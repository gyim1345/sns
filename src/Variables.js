export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
