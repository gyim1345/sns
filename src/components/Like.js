import React from 'react';
import Swal from 'sweetalert2';
import { css } from '@emotion/core';

import { getChangeLike } from '../apis/post';
import LikeSvg from '../svgIcons/LikeSvg';
import LikeFilledSvg from '../svgIcons/LikeFilledSvg';

const Like = ({ posting, setPosting, postingAll, currentUser }) => {
  const likeTrue = posting.like.includes(currentUser);
  const changeLikeOnClick = async () => {
    try {
      const response = await getChangeLike(posting);
      setPosting([response]);
      const index = postingAll.findIndex(it => posting.id === it.id);
      postingAll[index] = response;
      setPosting(postingAll);
    } catch (e) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Internal Error'
      });
    }
  };

  return (
    <div>
      <div onClick={changeLikeOnClick} css={[likeDiv]}></div>

      {likeTrue ? <LikeFilledSvg /> : <LikeSvg />}
      <div css={[margin1]}>
        좋아요 &nbsp;
        {posting.like.length}개
      </div>
    </div>
  );
};

const likeDiv = css`
  left: 16px;
  padding: 10px 12px 11px 12px;
  position: absolute;
`;

const margin1 = css`
  position: absolute;
  margin: -2px 0px 5px 16px;
  font-size: 14px;
  font-weight: bold;
`;

// Like.propTypes = {
//   currentUser: PropTypes.string,
//   posting: PropTypes.exact({
//     id: PropTypes.number,
//     title: PropTypes.string,
//     imageUrl: PropTypes.string,
//     userName: PropTypes.string,
//     like: PropTypes.arrayOf(PropTypes.string)
//   }),
//   postingAll: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
//   comments: PropTypes.arrayOf(PropTypes.object),
//   setPosting: PropTypes.elementType,
//   addComment: PropTypes.elementType,
//   onChangeComment: PropTypes.elementType,
//   commentAPI: PropTypes.oneOfType([PropTypes.array]),
//   setCommentAPI: PropTypes.elementType
// };

export default Like;
