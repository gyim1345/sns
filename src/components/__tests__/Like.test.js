import React from 'react';
import { mount } from 'enzyme';
import Like from '../Like';
import { post, posts } from '../../Variables';
import LikeSvg from '../../svgIcons/LikeSvg';

describe('<Like />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(<Like posting={post} postingAll={posts} />);

    expect(wrapper.props().posting).toBe(post);
    expect(wrapper.props().postingAll).toBe(posts);
    expect(wrapper.html()).toMatch('<div');
    expect(wrapper.text()).toMatch('좋아요');
    expect(wrapper.text()).toMatch(`${post.like.length}개`);
    expect(wrapper).toMatchSnapshot();
  });
});
