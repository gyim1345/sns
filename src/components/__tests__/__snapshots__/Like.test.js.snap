// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Like /> 스냅샷 비교 1`] = `
<Like
  posting={
    Object {
      "_id": "5e8ec52d9bb4913b5a844423",
      "id": 1,
      "imageUrl": "http://localhost:3000/static/images/1.jpg",
      "like": Array [
        "gibong@gmail.com",
      ],
      "tag": Array [
        "landscape",
      ],
      "title": "posting with id 1",
      "userName": "gibong@gmail.com",
    }
  }
  postingAll={
    Array [
      Object {
        "_id": "5e8ec52d9bb4913b5a844423",
        "id": 1,
        "imageUrl": "http://localhost:3000/static/images/1.jpg",
        "like": Array [
          "gibong@gmail.com",
        ],
        "tag": Array [
          "landscape",
        ],
        "title": "posting with id 1",
        "userName": "gibong@gmail.com",
      },
    ]
  }
>
  <div>
    <EmotionCssPropInternal
      __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Like"
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
      css={
        Object {
          "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxpa2UuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBdUNtQiIsImZpbGUiOiJMaWtlLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBTd2FsIGZyb20gJ3N3ZWV0YWxlcnQyJztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuXG5pbXBvcnQgeyBnZXRDaGFuZ2VMaWtlIH0gZnJvbSAnLi4vYXBpcy9wb3N0JztcbmltcG9ydCBMaWtlU3ZnIGZyb20gJy4uL3N2Z0ljb25zL0xpa2VTdmcnO1xuaW1wb3J0IExpa2VGaWxsZWRTdmcgZnJvbSAnLi4vc3ZnSWNvbnMvTGlrZUZpbGxlZFN2Zyc7XG5cbmNvbnN0IExpa2UgPSAoeyBwb3N0aW5nLCBzZXRQb3N0aW5nLCBwb3N0aW5nQWxsLCBjdXJyZW50VXNlciB9KSA9PiB7XG4gIGNvbnN0IGxpa2VUcnVlID0gcG9zdGluZy5saWtlLmluY2x1ZGVzKGN1cnJlbnRVc2VyKTtcbiAgY29uc3QgY2hhbmdlTGlrZU9uQ2xpY2sgPSBhc3luYyAoKSA9PiB7XG4gICAgdHJ5IHtcbiAgICAgIGNvbnN0IHJlc3BvbnNlID0gYXdhaXQgZ2V0Q2hhbmdlTGlrZShwb3N0aW5nKTtcbiAgICAgIHNldFBvc3RpbmcoW3Jlc3BvbnNlXSk7XG4gICAgICBjb25zdCBpbmRleCA9IHBvc3RpbmdBbGwuZmluZEluZGV4KGl0ID0+IHBvc3RpbmcuaWQgPT09IGl0LmlkKTtcbiAgICAgIHBvc3RpbmdBbGxbaW5kZXhdID0gcmVzcG9uc2U7XG4gICAgICBzZXRQb3N0aW5nKHBvc3RpbmdBbGwpO1xuICAgIH0gY2F0Y2ggKGUpIHtcbiAgICAgIFN3YWwuZmlyZSh7XG4gICAgICAgIGljb246ICdlcnJvcicsXG4gICAgICAgIHRpdGxlOiAnT29wcy4uLicsXG4gICAgICAgIHRleHQ6ICdJbnRlcm5hbCBFcnJvcidcbiAgICAgIH0pO1xuICAgIH1cbiAgfTtcblxuICByZXR1cm4gKFxuICAgIDxkaXY+XG4gICAgICA8ZGl2IG9uQ2xpY2s9e2NoYW5nZUxpa2VPbkNsaWNrfSBjc3M9e1tsaWtlRGl2XX0+PC9kaXY+XG5cbiAgICAgIHtsaWtlVHJ1ZSA/IDxMaWtlRmlsbGVkU3ZnIC8+IDogPExpa2VTdmcgLz59XG4gICAgICA8ZGl2IGNzcz17W21hcmdpbjFdfT5cbiAgICAgICAg7KKL7JWE7JqUICZuYnNwO1xuICAgICAgICB7cG9zdGluZy5saWtlLmxlbmd0aH3qsJxcbiAgICAgIDwvZGl2PlxuICAgIDwvZGl2PlxuICApO1xufTtcblxuY29uc3QgbGlrZURpdiA9IGNzc2BcbiAgbGVmdDogMTZweDtcbiAgcGFkZGluZzogMTBweCAxMnB4IDExcHggMTJweDtcbiAgcG9zaXRpb246IGFic29sdXRlO1xuYDtcblxuY29uc3QgbWFyZ2luMSA9IGNzc2BcbiAgcG9zaXRpb246IGFic29sdXRlO1xuICBtYXJnaW46IC0ycHggMHB4IDVweCAxNnB4O1xuICBmb250LXNpemU6IDE0cHg7XG4gIGZvbnQtd2VpZ2h0OiBib2xkO1xuYDtcblxuLy8gTGlrZS5wcm9wVHlwZXMgPSB7XG4vLyAgIGN1cnJlbnRVc2VyOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICBwb3N0aW5nOiBQcm9wVHlwZXMuZXhhY3Qoe1xuLy8gICAgIGlkOiBQcm9wVHlwZXMubnVtYmVyLFxuLy8gICAgIHRpdGxlOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIGltYWdlVXJsOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIHVzZXJOYW1lOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIGxpa2U6IFByb3BUeXBlcy5hcnJheU9mKFByb3BUeXBlcy5zdHJpbmcpXG4vLyAgIH0pLFxuLy8gICBwb3N0aW5nQWxsOiBQcm9wVHlwZXMub25lT2ZUeXBlKFtQcm9wVHlwZXMub2JqZWN0LCBQcm9wVHlwZXMuYXJyYXldKSxcbi8vICAgY29tbWVudHM6IFByb3BUeXBlcy5hcnJheU9mKFByb3BUeXBlcy5vYmplY3QpLFxuLy8gICBzZXRQb3N0aW5nOiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGFkZENvbW1lbnQ6IFByb3BUeXBlcy5lbGVtZW50VHlwZSxcbi8vICAgb25DaGFuZ2VDb21tZW50OiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGNvbW1lbnRBUEk6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5hcnJheV0pLFxuLy8gICBzZXRDb21tZW50QVBJOiBQcm9wVHlwZXMuZWxlbWVudFR5cGVcbi8vIH07XG5cbmV4cG9ydCBkZWZhdWx0IExpa2U7XG4iXX0= */",
          "name": "1lqxvv-likeDiv-Like",
          "next": undefined,
          "styles": "left:16px;padding:10px 12px 11px 12px;position:absolute;;label:likeDiv;;;label:Like;/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxpa2UuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBNEJ1QyIsImZpbGUiOiJMaWtlLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBTd2FsIGZyb20gJ3N3ZWV0YWxlcnQyJztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuXG5pbXBvcnQgeyBnZXRDaGFuZ2VMaWtlIH0gZnJvbSAnLi4vYXBpcy9wb3N0JztcbmltcG9ydCBMaWtlU3ZnIGZyb20gJy4uL3N2Z0ljb25zL0xpa2VTdmcnO1xuaW1wb3J0IExpa2VGaWxsZWRTdmcgZnJvbSAnLi4vc3ZnSWNvbnMvTGlrZUZpbGxlZFN2Zyc7XG5cbmNvbnN0IExpa2UgPSAoeyBwb3N0aW5nLCBzZXRQb3N0aW5nLCBwb3N0aW5nQWxsLCBjdXJyZW50VXNlciB9KSA9PiB7XG4gIGNvbnN0IGxpa2VUcnVlID0gcG9zdGluZy5saWtlLmluY2x1ZGVzKGN1cnJlbnRVc2VyKTtcbiAgY29uc3QgY2hhbmdlTGlrZU9uQ2xpY2sgPSBhc3luYyAoKSA9PiB7XG4gICAgdHJ5IHtcbiAgICAgIGNvbnN0IHJlc3BvbnNlID0gYXdhaXQgZ2V0Q2hhbmdlTGlrZShwb3N0aW5nKTtcbiAgICAgIHNldFBvc3RpbmcoW3Jlc3BvbnNlXSk7XG4gICAgICBjb25zdCBpbmRleCA9IHBvc3RpbmdBbGwuZmluZEluZGV4KGl0ID0+IHBvc3RpbmcuaWQgPT09IGl0LmlkKTtcbiAgICAgIHBvc3RpbmdBbGxbaW5kZXhdID0gcmVzcG9uc2U7XG4gICAgICBzZXRQb3N0aW5nKHBvc3RpbmdBbGwpO1xuICAgIH0gY2F0Y2ggKGUpIHtcbiAgICAgIFN3YWwuZmlyZSh7XG4gICAgICAgIGljb246ICdlcnJvcicsXG4gICAgICAgIHRpdGxlOiAnT29wcy4uLicsXG4gICAgICAgIHRleHQ6ICdJbnRlcm5hbCBFcnJvcidcbiAgICAgIH0pO1xuICAgIH1cbiAgfTtcblxuICByZXR1cm4gKFxuICAgIDxkaXY+XG4gICAgICA8ZGl2IG9uQ2xpY2s9e2NoYW5nZUxpa2VPbkNsaWNrfSBjc3M9e1tsaWtlRGl2XX0+PC9kaXY+XG5cbiAgICAgIHtsaWtlVHJ1ZSA/IDxMaWtlRmlsbGVkU3ZnIC8+IDogPExpa2VTdmcgLz59XG4gICAgICA8ZGl2IGNzcz17W21hcmdpbjFdfT5cbiAgICAgICAg7KKL7JWE7JqUICZuYnNwO1xuICAgICAgICB7cG9zdGluZy5saWtlLmxlbmd0aH3qsJxcbiAgICAgIDwvZGl2PlxuICAgIDwvZGl2PlxuICApO1xufTtcblxuY29uc3QgbGlrZURpdiA9IGNzc2BcbiAgbGVmdDogMTZweDtcbiAgcGFkZGluZzogMTBweCAxMnB4IDExcHggMTJweDtcbiAgcG9zaXRpb246IGFic29sdXRlO1xuYDtcblxuY29uc3QgbWFyZ2luMSA9IGNzc2BcbiAgcG9zaXRpb246IGFic29sdXRlO1xuICBtYXJnaW46IC0ycHggMHB4IDVweCAxNnB4O1xuICBmb250LXNpemU6IDE0cHg7XG4gIGZvbnQtd2VpZ2h0OiBib2xkO1xuYDtcblxuLy8gTGlrZS5wcm9wVHlwZXMgPSB7XG4vLyAgIGN1cnJlbnRVc2VyOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICBwb3N0aW5nOiBQcm9wVHlwZXMuZXhhY3Qoe1xuLy8gICAgIGlkOiBQcm9wVHlwZXMubnVtYmVyLFxuLy8gICAgIHRpdGxlOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIGltYWdlVXJsOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIHVzZXJOYW1lOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIGxpa2U6IFByb3BUeXBlcy5hcnJheU9mKFByb3BUeXBlcy5zdHJpbmcpXG4vLyAgIH0pLFxuLy8gICBwb3N0aW5nQWxsOiBQcm9wVHlwZXMub25lT2ZUeXBlKFtQcm9wVHlwZXMub2JqZWN0LCBQcm9wVHlwZXMuYXJyYXldKSxcbi8vICAgY29tbWVudHM6IFByb3BUeXBlcy5hcnJheU9mKFByb3BUeXBlcy5vYmplY3QpLFxuLy8gICBzZXRQb3N0aW5nOiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGFkZENvbW1lbnQ6IFByb3BUeXBlcy5lbGVtZW50VHlwZSxcbi8vICAgb25DaGFuZ2VDb21tZW50OiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGNvbW1lbnRBUEk6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5hcnJheV0pLFxuLy8gICBzZXRDb21tZW50QVBJOiBQcm9wVHlwZXMuZWxlbWVudFR5cGVcbi8vIH07XG5cbmV4cG9ydCBkZWZhdWx0IExpa2U7XG4iXX0= */",
          "toString": [Function],
        }
      }
      onClick={[Function]}
    >
      <div
        className="css-1lqxvv-likeDiv-Like"
        onClick={[Function]}
      />
    </EmotionCssPropInternal>
    <LikeSvg>
      <svg
        heigth="24"
        id="increaseLikeSvg"
        style={
          Object {
            "marginBottom": "5px",
            "marginLeft": "16px",
            "marginRight": "15px",
          }
        }
        type="button"
        viewBox="0 0 50 50"
        width="24"
      >
        <path
          className="heart"
          d="M 34.3 3.5 C 27.2 3.5 24 8.8 24 8.8 s -3.2 -5.3 -10.3 -5.3 C 6.4 3.5 0.5 9.9 0.5 17.8 s 6.1 12.4 12.2 17.8 c 9.2 8.2 9.8 8.9 11.3 8.9 s 2.1 -0.7 11.3 -8.9 c 6.2 -5.5 12.2 -10 12.2 -17.8 c 0 -7.9 -5.9 -14.3 -13.2 -14.3 Z m -1 29.8 c -5.4 4.8 -8.3 7.5 -9.3 8.1 c -1 -0.7 -4.6 -3.9 -9.3 -8.1 c -5.5 -4.9 -11.2 -9 -11.2 -15.6 c 0 -6.2 4.6 -11.3 10.2 -11.3 c 4.1 0 6.3 2 7.9 4.2 c 3.6 5.1 1.2 5.1 4.8 0 c 1.6 -2.2 3.8 -4.2 7.9 -4.2 c 5.6 0 10.2 5.1 10.2 11.3 c 0 6.7 -5.7 10.8 -11.2 15.6 Z"
        />
      </svg>
    </LikeSvg>
    <EmotionCssPropInternal
      __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Like"
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
      css={
        Object {
          "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxpa2UuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBNkNtQiIsImZpbGUiOiJMaWtlLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBTd2FsIGZyb20gJ3N3ZWV0YWxlcnQyJztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuXG5pbXBvcnQgeyBnZXRDaGFuZ2VMaWtlIH0gZnJvbSAnLi4vYXBpcy9wb3N0JztcbmltcG9ydCBMaWtlU3ZnIGZyb20gJy4uL3N2Z0ljb25zL0xpa2VTdmcnO1xuaW1wb3J0IExpa2VGaWxsZWRTdmcgZnJvbSAnLi4vc3ZnSWNvbnMvTGlrZUZpbGxlZFN2Zyc7XG5cbmNvbnN0IExpa2UgPSAoeyBwb3N0aW5nLCBzZXRQb3N0aW5nLCBwb3N0aW5nQWxsLCBjdXJyZW50VXNlciB9KSA9PiB7XG4gIGNvbnN0IGxpa2VUcnVlID0gcG9zdGluZy5saWtlLmluY2x1ZGVzKGN1cnJlbnRVc2VyKTtcbiAgY29uc3QgY2hhbmdlTGlrZU9uQ2xpY2sgPSBhc3luYyAoKSA9PiB7XG4gICAgdHJ5IHtcbiAgICAgIGNvbnN0IHJlc3BvbnNlID0gYXdhaXQgZ2V0Q2hhbmdlTGlrZShwb3N0aW5nKTtcbiAgICAgIHNldFBvc3RpbmcoW3Jlc3BvbnNlXSk7XG4gICAgICBjb25zdCBpbmRleCA9IHBvc3RpbmdBbGwuZmluZEluZGV4KGl0ID0+IHBvc3RpbmcuaWQgPT09IGl0LmlkKTtcbiAgICAgIHBvc3RpbmdBbGxbaW5kZXhdID0gcmVzcG9uc2U7XG4gICAgICBzZXRQb3N0aW5nKHBvc3RpbmdBbGwpO1xuICAgIH0gY2F0Y2ggKGUpIHtcbiAgICAgIFN3YWwuZmlyZSh7XG4gICAgICAgIGljb246ICdlcnJvcicsXG4gICAgICAgIHRpdGxlOiAnT29wcy4uLicsXG4gICAgICAgIHRleHQ6ICdJbnRlcm5hbCBFcnJvcidcbiAgICAgIH0pO1xuICAgIH1cbiAgfTtcblxuICByZXR1cm4gKFxuICAgIDxkaXY+XG4gICAgICA8ZGl2IG9uQ2xpY2s9e2NoYW5nZUxpa2VPbkNsaWNrfSBjc3M9e1tsaWtlRGl2XX0+PC9kaXY+XG5cbiAgICAgIHtsaWtlVHJ1ZSA/IDxMaWtlRmlsbGVkU3ZnIC8+IDogPExpa2VTdmcgLz59XG4gICAgICA8ZGl2IGNzcz17W21hcmdpbjFdfT5cbiAgICAgICAg7KKL7JWE7JqUICZuYnNwO1xuICAgICAgICB7cG9zdGluZy5saWtlLmxlbmd0aH3qsJxcbiAgICAgIDwvZGl2PlxuICAgIDwvZGl2PlxuICApO1xufTtcblxuY29uc3QgbGlrZURpdiA9IGNzc2BcbiAgbGVmdDogMTZweDtcbiAgcGFkZGluZzogMTBweCAxMnB4IDExcHggMTJweDtcbiAgcG9zaXRpb246IGFic29sdXRlO1xuYDtcblxuY29uc3QgbWFyZ2luMSA9IGNzc2BcbiAgcG9zaXRpb246IGFic29sdXRlO1xuICBtYXJnaW46IC0ycHggMHB4IDVweCAxNnB4O1xuICBmb250LXNpemU6IDE0cHg7XG4gIGZvbnQtd2VpZ2h0OiBib2xkO1xuYDtcblxuLy8gTGlrZS5wcm9wVHlwZXMgPSB7XG4vLyAgIGN1cnJlbnRVc2VyOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICBwb3N0aW5nOiBQcm9wVHlwZXMuZXhhY3Qoe1xuLy8gICAgIGlkOiBQcm9wVHlwZXMubnVtYmVyLFxuLy8gICAgIHRpdGxlOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIGltYWdlVXJsOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIHVzZXJOYW1lOiBQcm9wVHlwZXMuc3RyaW5nLFxuLy8gICAgIGxpa2U6IFByb3BUeXBlcy5hcnJheU9mKFByb3BUeXBlcy5zdHJpbmcpXG4vLyAgIH0pLFxuLy8gICBwb3N0aW5nQWxsOiBQcm9wVHlwZXMub25lT2ZUeXBlKFtQcm9wVHlwZXMub2JqZWN0LCBQcm9wVHlwZXMuYXJyYXldKSxcbi8vICAgY29tbWVudHM6IFByb3BUeXBlcy5hcnJheU9mKFByb3BUeXBlcy5vYmplY3QpLFxuLy8gICBzZXRQb3N0aW5nOiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGFkZENvbW1lbnQ6IFByb3BUeXBlcy5lbGVtZW50VHlwZSxcbi8vICAgb25DaGFuZ2VDb21tZW50OiBQcm9wVHlwZXMuZWxlbWVudFR5cGUsXG4vLyAgIGNvbW1lbnRBUEk6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5hcnJheV0pLFxuLy8gICBzZXRDb21tZW50QVBJOiBQcm9wVHlwZXMuZWxlbWVudFR5cGVcbi8vIH07XG5cbmV4cG9ydCBkZWZhdWx0IExpa2U7XG4iXX0= */",
          "name": "1l1qdgu-margin1-Like",
          "next": undefined,
          "styles": "position:absolute;margin:-2px 0px 5px 16px;font-size:14px;font-weight:bold;;label:margin1;;;label:Like;/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxpa2UuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBK0JXIiwiZmlsZSI6Ikxpa2UuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IFN3YWwgZnJvbSAnc3dlZXRhbGVydDInO1xuaW1wb3J0IHsgY3NzIH0gZnJvbSAnQGVtb3Rpb24vY29yZSc7XG5cbmltcG9ydCB7IGdldENoYW5nZUxpa2UgfSBmcm9tICcuLi9hcGlzL3Bvc3QnO1xuaW1wb3J0IExpa2VTdmcgZnJvbSAnLi4vc3ZnSWNvbnMvTGlrZVN2Zyc7XG5pbXBvcnQgTGlrZUZpbGxlZFN2ZyBmcm9tICcuLi9zdmdJY29ucy9MaWtlRmlsbGVkU3ZnJztcblxuY29uc3QgTGlrZSA9ICh7IHBvc3RpbmcsIHNldFBvc3RpbmcsIHBvc3RpbmdBbGwsIGN1cnJlbnRVc2VyIH0pID0+IHtcbiAgY29uc3QgbGlrZVRydWUgPSBwb3N0aW5nLmxpa2UuaW5jbHVkZXMoY3VycmVudFVzZXIpO1xuICBjb25zdCBjaGFuZ2VMaWtlT25DbGljayA9IGFzeW5jICgpID0+IHtcbiAgICB0cnkge1xuICAgICAgY29uc3QgcmVzcG9uc2UgPSBhd2FpdCBnZXRDaGFuZ2VMaWtlKHBvc3RpbmcpO1xuICAgICAgc2V0UG9zdGluZyhbcmVzcG9uc2VdKTtcbiAgICAgIGNvbnN0IGluZGV4ID0gcG9zdGluZ0FsbC5maW5kSW5kZXgoaXQgPT4gcG9zdGluZy5pZCA9PT0gaXQuaWQpO1xuICAgICAgcG9zdGluZ0FsbFtpbmRleF0gPSByZXNwb25zZTtcbiAgICAgIHNldFBvc3RpbmcocG9zdGluZ0FsbCk7XG4gICAgfSBjYXRjaCAoZSkge1xuICAgICAgU3dhbC5maXJlKHtcbiAgICAgICAgaWNvbjogJ2Vycm9yJyxcbiAgICAgICAgdGl0bGU6ICdPb3BzLi4uJyxcbiAgICAgICAgdGV4dDogJ0ludGVybmFsIEVycm9yJ1xuICAgICAgfSk7XG4gICAgfVxuICB9O1xuXG4gIHJldHVybiAoXG4gICAgPGRpdj5cbiAgICAgIDxkaXYgb25DbGljaz17Y2hhbmdlTGlrZU9uQ2xpY2t9IGNzcz17W2xpa2VEaXZdfT48L2Rpdj5cblxuICAgICAge2xpa2VUcnVlID8gPExpa2VGaWxsZWRTdmcgLz4gOiA8TGlrZVN2ZyAvPn1cbiAgICAgIDxkaXYgY3NzPXtbbWFyZ2luMV19PlxuICAgICAgICDsoovslYTsmpQgJm5ic3A7XG4gICAgICAgIHtwb3N0aW5nLmxpa2UubGVuZ3RofeqwnFxuICAgICAgPC9kaXY+XG4gICAgPC9kaXY+XG4gICk7XG59O1xuXG5jb25zdCBsaWtlRGl2ID0gY3NzYFxuICBsZWZ0OiAxNnB4O1xuICBwYWRkaW5nOiAxMHB4IDEycHggMTFweCAxMnB4O1xuICBwb3NpdGlvbjogYWJzb2x1dGU7XG5gO1xuXG5jb25zdCBtYXJnaW4xID0gY3NzYFxuICBwb3NpdGlvbjogYWJzb2x1dGU7XG4gIG1hcmdpbjogLTJweCAwcHggNXB4IDE2cHg7XG4gIGZvbnQtc2l6ZTogMTRweDtcbiAgZm9udC13ZWlnaHQ6IGJvbGQ7XG5gO1xuXG4vLyBMaWtlLnByb3BUeXBlcyA9IHtcbi8vICAgY3VycmVudFVzZXI6IFByb3BUeXBlcy5zdHJpbmcsXG4vLyAgIHBvc3Rpbmc6IFByb3BUeXBlcy5leGFjdCh7XG4vLyAgICAgaWQ6IFByb3BUeXBlcy5udW1iZXIsXG4vLyAgICAgdGl0bGU6IFByb3BUeXBlcy5zdHJpbmcsXG4vLyAgICAgaW1hZ2VVcmw6IFByb3BUeXBlcy5zdHJpbmcsXG4vLyAgICAgdXNlck5hbWU6IFByb3BUeXBlcy5zdHJpbmcsXG4vLyAgICAgbGlrZTogUHJvcFR5cGVzLmFycmF5T2YoUHJvcFR5cGVzLnN0cmluZylcbi8vICAgfSksXG4vLyAgIHBvc3RpbmdBbGw6IFByb3BUeXBlcy5vbmVPZlR5cGUoW1Byb3BUeXBlcy5vYmplY3QsIFByb3BUeXBlcy5hcnJheV0pLFxuLy8gICBjb21tZW50czogUHJvcFR5cGVzLmFycmF5T2YoUHJvcFR5cGVzLm9iamVjdCksXG4vLyAgIHNldFBvc3Rpbmc6IFByb3BUeXBlcy5lbGVtZW50VHlwZSxcbi8vICAgYWRkQ29tbWVudDogUHJvcFR5cGVzLmVsZW1lbnRUeXBlLFxuLy8gICBvbkNoYW5nZUNvbW1lbnQ6IFByb3BUeXBlcy5lbGVtZW50VHlwZSxcbi8vICAgY29tbWVudEFQSTogUHJvcFR5cGVzLm9uZU9mVHlwZShbUHJvcFR5cGVzLmFycmF5XSksXG4vLyAgIHNldENvbW1lbnRBUEk6IFByb3BUeXBlcy5lbGVtZW50VHlwZVxuLy8gfTtcblxuZXhwb3J0IGRlZmF1bHQgTGlrZTtcbiJdfQ== */",
          "toString": [Function],
        }
      }
    >
      <div
        className="css-1l1qdgu-margin1-Like"
      >
        좋아요  
        1
        개
      </div>
    </EmotionCssPropInternal>
  </div>
</Like>
`;
